utils/obsproc/NetCDFToIodaConverter2D.h

```diff
@@ -0,0 +1,318 @@
+      longitude_(location_), latitude_(location_), datetime_(location_), channelNumber_(channel_),
+      obsVal_(location_ * channel_),
+      obsError_(location_ * channel_),
+      preQc_(location_ * channel_),
+      floatMetadata_(location_, fmnames.size()),
+      floatMetadataName_(fmnames),
+      intMetadata_(location_, imnames.size()),
+      intMetadataName_(imnames)
+    {
+      oops::Log::trace() << "IodaVars2d::IodaVars2d created." << std::endl;
+    }
+
+    // Append an other instance of IodaVars
+    void append(const IodaVars2d& other) {
+      // Check if the two instances can be concatenated
+      ASSERT(nVars_ == other.nVars_);
+      ASSERT(nfMetadata_ == other.nfMetadata_);
+      ASSERT(niMetadata_ == other.niMetadata_);
+      ASSERT(floatMetadataName_ == floatMetadataName_);
+      ASSERT(intMetadataName_ == intMetadataName_);
+
+      // Concatenate Eigen arrays and vectors
+      longitude_.conservativeResize(location_ + other.location_);
+      latitude_.conservativeResize(location_ + other.location_);
+      datetime_.conservativeResize(location_ + other.location_);
+      obsVal_.conservativeResize(location_ * channel_ + other.location_ * other.channel_);
+      obsError_.conservativeResize(location_ * channel_ + other.location_ * other.channel_);
+      preQc_.conservativeResize(location_ * channel_ + other.location_ * other.channel_);
+      floatMetadata_.conservativeResize(location_ + other.location_, nfMetadata_);
+      intMetadata_.conservativeResize(location_ + other.location_, niMetadata_);
+
+      // Copy data from the 'other' object to this object
+      longitude_.tail(other.location_) = other.longitude_;
+      latitude_.tail(other.location_) = other.latitude_;
+      datetime_.tail(other.location_) = other.datetime_;
+      obsVal_.tail(other.location_ * other.channel_) = other.obsVal_;
+      obsError_.tail(other.location_ * other.channel_) = other.obsError_;
+      preQc_.tail(other.location_ * other.channel_) = other.preQc_;
+      floatMetadata_.bottomRows(other.location_) = other.floatMetadata_;
+      intMetadata_.bottomRows(other.location_) = other.intMetadata_;
+
+      // Update obs count
+      location_ += other.location_;
+      oops::Log::trace() << "IodaVars2d::IodaVars2d done appending." << std::endl;
+    }
+  };
+
+  // Base class for the converters
+  class NetCDFToIodaConverter2d {
+   public:
+    // Constructor: Stores the configuration as a data members
+    explicit NetCDFToIodaConverter2d(const eckit::Configuration & fullConfig,
+                                   const eckit::mpi::Comm & comm): comm_(comm),
+                                                                   fullConfig_(fullConfig) {
+      // time window info
+      std::string winbegin;
+      std::string winend;
+      fullConfig.get("window begin", winbegin);
+      fullConfig.get("window end", winend);
+      windowBegin_ = util::DateTime(winbegin);
```

Comment:
window info is probably not used, at least we don't and I suspect it's the same here.